### PR TITLE
adding ability to fetch all objects if you don't include the ids

### PR DESCRIPTION
**Pull request by erchaves**
Also including a convenient debug script in package.json

### CODE REVIEW

**Comment by erchaves** [comment]
@jedahan Feel free to adjust if this doesn't feel like the right syntax or approach.

File: src/emuseum.js [hunk 1]
```diff
@@ -13,33 +46,19 @@ module.exports = ({ emuseumKey }) => {
 
   return {
     async getObjects ({ ids }) {
-      const rawObjects = await this.getRawObjects({ ids })
+      let rawObjects;
```
Comment (jedahan) [inline src/emuseum.js]: maybe run through `npm run format`

I wonder how easy it would be to add that as a hook to PRs...

File: src/emuseum.js [hunk 2]
```diff
@@ -13,33 +46,19 @@ module.exports = ({ emuseumKey }) => {
...
-          id,
-          title,
-          maker,
-          medium,
-          classification,
-          creditline,
-          dimensions,
-          raw
-        }
+        return parseRawObject(raw);
```
Comment (jedahan) [inline src/emuseum.js]: this can be consolidated to
```js
const objects = rawObject.map(parseRawObject)
```

File: src/emuseum.js [hunk 3]
```diff
@@ -54,6 +73,15 @@ module.exports = ({ emuseumKey }) => {
         })
       )
       return response
+    },
+    async getRawObjectsAll () {
+      if (!emuseumKey) throw MISSING_EMUSEUM_KEY
+
+      const uri = `http://egallery.williams.edu/objects/json?key=${emuseumKey}`
```
Comment (jedahan) [inline src/emuseum.js:78]: confirmed this gives all object? unpaginated? thats pretty damn nice

I wonder if the performance is good enough if we just delegate to grabbing all, and we use in memory cache to search ourselves after the first hit for queries with ids. either way, thats for another PR.

File: src/emuseum.js [hunk 4]
```diff
@@ -2,6 +2,39 @@ module.exports = ({ emuseumKey }) => {
...
+      title: { value: title },
+      medium,
+      classification,
+      creditline,
+      dimensions,
+      primaryMaker,
+      primaryMedia,
+      people
+    } = raw
+
```
Comment (jedahan) [inline src/emuseum.js:17]: for consistency sake we can make id and title like the lines below

```js
id = id && id.value
title = title && title.value
```
we might want an `|| null`, or a more generic / strict filter in the future

**Review by jedahan — CHANGES_REQUESTED** [review]
3 recommended changes, 2 thinking-out-loud ideas for future improvements

**Review by jedahan — APPROVED** [review]

File: src/emuseum.js (hunk 3 above)
Comment (erchaves) [reply src/emuseum.js:78]: It returns just a first page I believe. And I did a quick test of pagination which the api does support. So I'll add a todo to make this properly paginated

File: src/emuseum.js (hunk 4 above)
Comment (erchaves) [reply src/emuseum.js:17]: cool will do

File: src/emuseum.js (hunk 1 above)
Comment (erchaves) [reply src/emuseum.js]: ah nice, yeah ill do that.

File: src/emuseum.js (hunk 2 above)
Comment (erchaves) [reply src/emuseum.js]: 👍 will do

**Comment by erchaves** [comment]
Oops, i never got back to this.. I'll look at these changes today.

**Pull request merged**